src/transforms/dedupe.rs

```diff
@@ -0,0 +1,269 @@
+use super::Transform;
+use crate::{
+    event::{Event, Value},
+    runtime::TaskExecutor,
+    topology::config::{DataType, TransformConfig, TransformDescription},
+};
+use bytes::Bytes;
+use chrono::{DateTime, Utc};
+use lru::LruCache;
+use serde::{Deserialize, Serialize};
+use std::any::TypeId;
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct FieldMatchConfig {
+    #[serde(rename = "match")]
+    pub match_fields: Vec<Atom>,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct CacheConfig {
+    pub num_entries: usize,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct DedupeConfig {
+    #[serde(default = "default_field_match_config")]
+    pub fields: FieldMatchConfig,
+    #[serde(default = "default_cache_config")]
+    pub cache: CacheConfig,
+}
+
+fn default_cache_config() -> CacheConfig {
+    CacheConfig { num_entries: 5000 }
+}
+
+fn default_field_match_config() -> FieldMatchConfig {
+    FieldMatchConfig {
+        match_fields: vec!["timestamp".into()],
+    }
+}
+
+pub struct Dedupe {
+    config: DedupeConfig,
+    cache: LruCache<CacheEntry, bool>,
+}
+
+inventory::submit! {
+    TransformDescription::new_without_default::<DedupeConfig>("dedupe")
+}
+
+#[typetag::serde(name = "dedupe")]
+impl TransformConfig for DedupeConfig {
+    fn build(&self, _exec: TaskExecutor) -> crate::Result<Box<dyn Transform>> {
+        Ok(Box::new(Dedupe::new(
+            self.cache.num_entries,
+            self.fields.match_fields.clone(),
+        )))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "dedupe"
+    }
+}
+
+type CacheEntry = Vec<Option<(TypeId, Bytes)>>;
+
+fn type_id_for_value(val: &Value) -> TypeId {
+    match val {
+        Value::Bytes(_) => TypeId::of::<Bytes>(),
+        Value::Timestamp(_) => TypeId::of::<DateTime<Utc>>(),
+        Value::Integer(_) => TypeId::of::<i64>(),
+        Value::Float(_) => TypeId::of::<f64>(),
+        Value::Boolean(_) => TypeId::of::<bool>(),
+    }
+}
+
+impl Dedupe {
+    pub fn new(num_entries: usize, match_fields: Vec<Atom>) -> Self {
+        Self {
+            config: DedupeConfig {
+                fields: FieldMatchConfig { match_fields },
+                cache: CacheConfig { num_entries },
+            },
+            cache: LruCache::new(num_entries),
+        }
+    }
+
+    /**
+     * Takes in an Event and returns a new Event with only the fields that are being matched against.
```

Comment:
This comment is incorrect as it's returning a CacheEntry

Reply:
Oops, I fixed that locally already but must have forgotten to update the PR.  My bad